int32 x
int32 y
int32 width
int32 height
int32 image_width
int32 image_height
